### PR TITLE
Fix patches

### PR DESCRIPTION
**Pull request by clegoues**
Copied and pasted code from mostly the revert strategy.

### CODE REVIEW

File: src/repairchain/util.py [hunk 1]
```diff
@@ -16,9 +23,73 @@
...
+        pass
+    try:
+        logger.debug(f"restoring to: {restore_to}")
+        repo.git.reset(restore_to)
+    except repo.git.exc.GitCommandError:
+        logger.exception(f"failed to restore git repo to: {restore_to}")
+
+
+def apply_patch(project: Project, patch: Diff) -> None:
+    # make a commit consisting of only the minimized undo
```
Comment (ChrisTimperley) [inline src/repairchain/util.py:64]: should delete this comment

File: src/repairchain/util.py [hunk 2]
```diff
@@ -16,9 +23,73 @@
...
+            )
+
+
+def diff_to_git_diff(
+        project: Project,
+        patch: Diff,
+) -> str:
+    restore_to_head(project)
+    apply_patch(project, patch)
+    return str(project.repository.git.diff())
```
Comment (ChrisTimperley) [inline src/repairchain/util.py:90]: does `.diff()` return the right format for us?

File: src/repairchain/util.py (hunk 1 above)
Comment (clegoues) [reply src/repairchain/util.py:64]: yup done

File: src/repairchain/util.py [hunk 3]
```diff
@@ -16,9 +23,73 @@
...
+                stdin=subprocess.DEVNULL,
+            )
+
+
+def diff_to_git_diff(
+        project: Project,
+        patch: Diff,
+) -> str:
+    restore_to_head(project)
+    apply_patch(project, patch)
```
Comment (ChrisTimperley) [inline src/repairchain/util.py:89]: wrap this in a `try:` `finally:` that calls `restore_to_head`

File: src/repairchain/util.py (hunk 2 above)
Comment (clegoues) [reply src/repairchain/util.py:90]: I'm asking about this.  Will want more options to diff() too

**Comment by ChrisTimperley** [comment]
Resolved by adding "a/" and "b/" as prefixes to the diff

**Pull request closed without merging**